hydra-tx/src/Hydra/Tx/Fanout.hs

```diff
@@ -90,8 +91,12 @@ observeFanoutTx ::
 observeFanoutTx utxo tx = do
   let inputUTxO = resolveInputsUTxO utxo tx
   (headInput, headOutput) <- findTxOutByScript inputUTxO Head.validatorScript
+  let txOutputs = utxoFromTx tx
   headId <- findStateToken headOutput
   findRedeemerSpending tx headInput
     >>= \case
-      Head.Fanout{} -> pure FanoutObservation{headId}
+      Head.Fanout{numberOfFanoutOutputs, numberOfCommitOutputs, numberOfDecommitOutputs} -> do
+        let allOutputs = fromIntegral $ numberOfFanoutOutputs + numberOfCommitOutputs + numberOfDecommitOutputs
+        let fanoutUTxO = UTxO.fromPairs $ take allOutputs $ UTxO.pairs txOutputs
```

Comment:
Should: not create a `UTxO` above if we want a `[TxOut]` here

This is a bit fragile as it relies on the fact that the ordering by `TxId` is consistent through `utxoFromTx` and `pairs`. Instead, we could access the outputs directly here:
```suggestion
        let fanoutUTxO = UTxO.fromPairs $ zipWith (mkTxIn tx) [0..] $ take allOutputs $ txOuts' tx
```